src/ssz/tree_view/list_composite.zig

```diff
@@ -179,5 +180,29 @@ pub fn ListCompositeTreeView(comptime ST: type) type {
             errdefer self.base_view.pool.unref(length_node);
             try self.base_view.setChildNode(@enumFromInt(3), length_node);
         }
+
+        /// Serialize the tree view directly to binary data.
+        /// This commits any pending changes and then serializes from the merkle tree.
+        pub fn serialize(self: *Self) ![]u8 {
```

Comment:
I'm not sure if we really need this `serialize()` api
one of the pain point of lodestar is we have to allocate everywhere due to missing "*into()" apis, so with `clone()` we decided to only support "into()" api, also type only supported "serializeIntoBytes()"
then consumer needs to decide how they deal with memory: using stack vs heap, reuse memory etc..

I'd say let's drop the support for `serialize()` api, we can add it back in the future if it's really needed